Sources/AWSLambdaEvents/AppSync.swift

```diff
@@ -0,0 +1,110 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2017-2020 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public struct AppSync {
+	public struct Request: Codable {
+		public let arguments: [String: ArgumentValue]
+		
+		public enum ArgumentValue: Codable {
+			case string(String)
+			case dictionary([String: String])
+			
+			public init(from decoder: Decoder) throws {
+				let container = try decoder.singleValueContainer()
+				if let strValue = try? container.decode(String.self) {
+					self = .string(strValue)
+				} else if let dictionaryValue = try? container.decode([String: String].self) {
+					self = .dictionary(dictionaryValue)
+				} else {
+					throw DecodingError.dataCorruptedError(in: container, debugDescription: """
+					Unexpected AppSync argument.
+					Expected a String or a Dictionary.
+					""")
+				}
+			}
+			
+			public func encode(to encoder: Encoder) throws {
+				var container = encoder.singleValueContainer()
+				switch self {
+				case .dictionary(let array):
+					try container.encode(array)
+				case .string(let str):
+					try container.encode(str)
+				}
+			}
+		}
+		
+		public let info: Info
+		public struct Info: Codable {
+			public var selectionSetList: [String]
+			public var selectionSetGraphQL: String
+			public var parentTypeName: String
+			public var fieldName: String
+			public var variables: [String: String]
+		}
+		public let identity: Identity
```

Comment:
```suggestion
		public let identity: Identity?
```
In the case of `API_KEY` this property isn't populated